### DISCUSSION TITLE
Focal point functionality

### DISCUSSION THREAD

**Discussion by leventefabry-mof** [saucecontrol/PhotoSauce#104] [Q&A]
Hi,
I'm quite new to the MagicScaler package but I want to know how can I achieve a focal point functionality like in the ImageSharp package?

Thanks in advance!

**Comment by saucecontrol** [comment]
I'm not sure which specific functionality from ImageSharp you mean -- maybe some kind of feature detection?  

There's an open request for smart crop https://github.com/saucecontrol/PhotoSauce/issues/54 which has some discussion about why that kind of functionality is not a priority for MagicScaler.  In general, if a feature requires full materialization of an image into memory, you won't find it here.  That is a fundamental architectural difference between MagicScaler and ImageSharp, so you will find they do not have feature parity in those areas.

**Reply by leventefabry-mof** [reply]
By the focal point I mean you can set a x and y coordinate and that will be in focus if you use a crop. Anyway, I was able to solve the issue with Crop and Crop Basis and with a little calculation.

Thanks for your help!
